src/kOS.Safe/Execution/Stack.cs

```diff
@@ -9,25 +9,73 @@ namespace kOS.Safe.Execution
     public class Stack : IStack
     {
         private const int MAX_STACK_SIZE = 3000;
-        private readonly List<object> stack = new List<object>();
-        private int stackPointer = -1;
+        private readonly object[] stack = new object[MAX_STACK_SIZE];
+        private int count = 0;
+        private readonly object[] aboveStack = new object[MAX_STACK_SIZE];
```

Comment:
**We should stop calling this "above".**
If we're changing this into two different stacks, I think it we should also change the terminology.
This was called the "above" stack purely because of where it was located - above the other stack.
This terminology change should also affect all the other places the word "above" is used to refer to this stack, like PopAbove, PushAbove, aboveCount, etc.  There may also be mentions of this in comments trying to explain the "above stack", which should change too.

Ideas for new term: scopeStack, or callStack.

Reply:
**Note to self**: this will be a source of merge conflict for me:  On the subject of creating the stack so it stays permanently at MAX_STACK_SIZE all the time - I had a change in my clock cycle branch designed to do this already but it wasn't ready to post it - I was keeping it as a stack (not an array) but telling .net to pre-allocate the stack to the full capacity.  I will need to take care when updating that branch to develop because this will conflict for sure.

Reply:
ah yeah good point. i was just trying to keep things as similar as possible, but some clearer naming here would definitely make this more approachable in the future.